selene-lib/src/lints/roblox_complex_udim2_new.rs

```diff
@@ -0,0 +1,176 @@
+    arg_1: f32,
+    no_scale: bool,
+}
+
+impl Visitor for UDim2NewVisitor {
+    fn visit_function_call(&mut self, call: &ast::FunctionCall) {
+        if_chain::if_chain! {
+            if let ast::Prefix::Name(token) = call.prefix();
+            if token.token().to_string() == "UDim2";
+            let mut suffixes = call.suffixes().collect::<Vec<_>>();
+
+            if suffixes.len() == 2; // .new and ()
+            let call_suffix = suffixes.pop().unwrap();
+            let index_suffix = suffixes.pop().unwrap();
+
+            if let ast::Suffix::Index(ast::Index::Dot { name, .. }) = index_suffix;
+            if name.token().to_string() == "new";
+
+            if let ast::Suffix::Call(ast::Call::AnonymousCall(
+                ast::FunctionArgs::Parentheses { arguments, .. }
+            )) = call_suffix;
+
+            then {
+                let args_provided = arguments.len();
+                if args_provided != 4 {
+                    return;
+                }
+
+
+                let numbers_passed = arguments.iter().filter(|expression| {
+                    matches!(expression, ast::Expression::Number(_))
+                }).count();
+                if numbers_passed != 4 {
+                    return;
+                }
+
+                let mut iter = arguments.iter();
+                let x_scale = match iter.next().unwrap().to_string().parse::<f32>() {
+                    Ok(value) => value,
+                    Err(_) => return,
+                };
+                let x_offset = match iter.next().unwrap().to_string().parse::<f32>() {
+                    Ok(value) => value,
+                    Err(_) => return,
+                };
+                let y_scale = match iter.next().unwrap().to_string().parse::<f32>() {
+                    Ok(value) => value,
+                    Err(_) => return,
+                };
+                let y_offset = match iter.next().unwrap().to_string().parse::<f32>() {
+                    Ok(value) => value,
+                    Err(_) => return,
+                };
+
+                let no_scale = x_scale == 0.0 && y_scale == 0.0;
+                let no_offset = x_offset == 0.0 && y_offset == 0.0;
+
+                let arg_0;
+                let arg_1;
+                if no_scale && no_offset
+                {
+                    // Skip any lint
+                    return;
+                }
+                else if no_scale
+                {
+                    arg_0 = x_offset;
+                    arg_1 = y_offset;
+                }
+                else if no_offset {
+                    arg_0 = x_scale;
+                    arg_1 = y_scale;
+                }
+                else
+                {
+                    return;
+                }
+
+                self.args.push(UDim2ComplexArgs {
+                    call_range: range(call),
+                    arg_0: arg_0,
+                    arg_1: arg_1,
+                    no_scale: no_scale,
```

Comment:
```suggestion
                    arg_0,
                    arg_1,
                    no_scale,
```

Reply:
Nice! Unfortunately don't get to keep this with the refactor, but this is good to know.